### DISCUSSION TITLE
Can't create more than max_prepared_stmt_count

### DISCUSSION THREAD

**Discussion by webnoob** [prisma/prisma#12225] [Q&A] [answered]
Hi,

I've gone live with my product and generally things are running very well. Randomly throughout the day though, we'll end up getting this error: 

![image](https://user-images.githubusercontent.com/10404747/157472880-af7e37e1-b120-4665-8c28-ab8ae4e56a29.png)

I've seen this in development a number of times when I write a query which tries to get too much data at once but the problem is with this is that it's not tied to one specific query. When it happens, it then appears for EVERY query after that - even small ones. The only way to recover is to restart the mysql server.

Does anyone have any advice on how to track this down?

Also, if I can "catch it" and clear a buffer or something short term, that'd help a lot.

I'm using MYSQL and prisma-client-js.

EDIT: Ok, it seems that `show global status like 'Com_stmt_prepare%';` shows this crashes when the number hits around 40k (despite the max setting showing as 16382...).

When / how does Prisma deal with closing it's prepared statements because so far - I have far more prepared than closed.

**Comment by austincrim** [comment, marked as answer]
Hey @webnoob 👋,

You can configure the amount of prepared statements cached with the `statement_cache_size` query parameter in your connection string. I believe it defaults to 1000 so you could try decreasing that significantly, maybe to around 100 to start.

```
DATABASE_URL="postgresql://user:pass@myserver:5432/public?statement_cache_size=100"
```

40,000 does seem like a lot, though. How many servers or Prisma Client instances are connecting to your db? What does your connection pool setup look like?

**Reply by webnoob** [reply]
It turns out I had one slow running query causing almost all the issues. Also, the way I was counting the prepared statements was wrong so the correct query to do that is: `SELECT variable_value FROM performance_schema.global_status WHERE  VARIABLE_NAME = 'Prepared_stmt_count';`

Since sorting the query and looking at the correct value, things have generally been fine. I've had one instance where this limit was reached but I think that was a one off.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
